src/librustdoc/config.rs

```diff
@@ -159,7 +159,12 @@ crate struct Options {
     /// Whether doctests should emit unused externs
     crate json_unused_externs: bool,
 
+    // Options for scraping call sites from examples/ directory
+    /// Path to output file to write JSON of call sites. If this option is Some(..) then
+    /// the compiler will scrape examples and not generate documentation.
     crate scrape_examples: Option<PathBuf>,
+    /// Path to the root of the workspace, used to generate workspace-relative file paths.
+    crate workspace_root: Option<PathBuf>,
```

Comment:
Rustdoc does not currently have an idea of workspaces, I really hate to make cargo ideas part of rustdoc itself. Why does it need to be workspace relative instead of crate relative or something?

Reply:
The ultimate goal is to generate a URL to the source code for a given example. For instance, take [gtk-rs](https://github.com/gtk-rs/gtk-rs). It has examples in `/examples/`. Let's say we're documenting `/gtk/sys/` and there's a call to `gtk_sys::foo` in `/examples/overlay/main.rs`. The goal is to generate a URL to https://github.com/gtk-rs/gtk-rs/examples/overlay/main.rs .

I'm not sure how you accomplish this task without knowing the workspace. A crate-relative path to `main.rs` would be `overlay/main.rs`, which doesn't include the `examples` directory. 

Alternatively, we could avoid this task altogether by including source code for examples as a part of Rustdoc's generated files, and then linking to those files.

Reply:
The workspace is still not the thing you want, you want the relative path within the git repository. The workspace might not be at the root.

Yes, I agree generating the sources ourselves is a better solution; it was mentioned in the RFC too.

Reply:
Fixed in b23132d